File: code/ts/example/models/todo.ts
/// Modelo de dominio para un item ToDo de JSONPlaceholder.
export class ToDo {
  readonly id: number;
  readonly title: string;
  readonly isCompleted: boolean;

  private constructor(id: number, title: string, isCompleted: boolean) {
    this.id = id;
    this.title = title;
    this.isCompleted = isCompleted;
  }

  static fromJson(json: Record<string, unknown>): ToDo {
    return new ToDo(
      json['id'] as number,
      json['title'] as string,
      json['completed'] as boolean,
    );
  }

  toString(): string {
    return `ToDo(id: ${this.id}, title: "${this.title}", isCompleted: ${this.isCompleted})`;
  }
}

File: code/py/example/models/todo.py
"""Modelo de dominio para un item ToDo de JSONPlaceholder."""

from __future__ import annotations

from dataclasses import dataclass
from typing import Any


@dataclass(frozen=True, slots=True)
class ToDo:
    id: int
    title: str
    is_completed: bool

    @staticmethod
    def from_json(json: dict[str, Any]) -> ToDo:
        return ToDo(
            id=json["id"],
            title=json["title"],
            is_completed=json["completed"],
        )

    def __str__(self) -> str:
        return f'ToDo(id: {self.id}, title: "{self.title}", is_completed: {self.is_completed})'
